src/components/interfaces/HMI_API.xml

```diff
@@ -3281,6 +3300,22 @@
       <description>ID of application related to this RPC.</description>
     </param>
   </function>
+  <function name="SendHapticData" messagetype="request">
+    <description>Send the UI spatial data from MOBILE. This data will be utilized by the HMI to determine how and when haptic events should occur</description>
+    <param name="appID" type="Integer" mandatory="true">
+      <description>Internal ID of the application that requested this RPC.</description>
+    </param>
+    <param name="HapticSpatialData" type="Common.SpatialStruct" minsize="0" maxsize="1000" mandatory="false" array="true">
```

Comment:
Note: will likely need to be changed to `hapticSpatialData`, I have offered a change to the original proposal. https://github.com/smartdevicelink/sdl_evolution/pull/253